msgpack/_packer.pyx

```diff
@@ -224,27 +225,29 @@ cdef class Packer(object):
                 longval = o.code
                 rawval = o.data
                 L = len(o.data)
-                if L > (2**32)-1:
+                if L > ITEM_LIMIT:
                     raise PackValueError("EXT data is too large")
                 ret = msgpack_pack_ext(&self.pk, longval, L)
                 ret = msgpack_pack_raw_body(&self.pk, rawval, L)
             elif PyList_CheckExact(o) if strict_types else (PyTuple_Check(o) or PyList_Check(o)):
                 L = len(o)
-                if L > (2**32)-1:
+                if L > ITEM_LIMIT:
                     raise PackValueError("list is too large")
                 ret = msgpack_pack_array(&self.pk, L)
                 if ret == 0:
                     for v in o:
                         ret = self._pack(v, nest_limit-1)
                         if ret != 0: break
             elif PyMemoryView_Check(o):
+                if not self.pk.use_bin_type:
+                    raise ValueError("need use_bin_type=True to pack memoryview")
                 if PyObject_GetBuffer(o, &view, PyBUF_SIMPLE) != 0:
-                    raise PackValueError("could not get buffer for memoryview")
+                    raise ValueError("could not get buffer for memoryview")
```

Comment:
hmm, if i understood correctly, PackValueError is only deprecated.

That means, users of msgpack must now change their code to catch ValueError instead of PackValueError. But as long as they still might catch PackValueError, msgpack must not be changed to raise ValueError because PackValueError is more special than ValueError and thus "except PackValueError" won't catch ValueError.